tl.tl

```diff
@@ -4022,11 +4041,20 @@ local function fill_field_order(t: Type)
    end
 end
 
-local function require_module(module_name: string, lax: boolean, env: Env, result: Result): Type, boolean
+function tl.require_module(module_name: string, lax: boolean, env: Env, result: Result): Type, boolean
+   env = env or tl.init_env(lax)
    local modules = env.modules
    local loaded = env.loaded
 
    if modules[module_name] then
+      if result and not result.dependencies[module_name] then
+         -- TODO: should we store a map of module names to file names in Env to get rid of this redundant search?
```

Comment:
yes, I think so. We already have `loaded`, which is a map from filenames to Results, and `modules` which is a map from module names to types. A map of module names to file names would be essentially a join table between these two. 

However....

Don't we already have `filename` in the Type? I don't think all generated Type objects currently have that set, but we could ensure that everything returned by `require()` does. In that case, you could have your mapping from module name to filename by doing `env.modules[module_name].filename`.